pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1082,8 +1082,19 @@ cdef class _Timedelta(timedelta):
             #  non-invariant behavior.
             #  see GH#44504
             return hash(self.value)
-        else:
+        elif self._reso == NPY_FR_ns:
             return timedelta.__hash__(self)
+        else:
+            # We want to ensure that two equivalent Timedelta objects
+            #  have the same hash.  So we try downcasting to the next-lowest
+            #  resolution.
+            try:
+                obj = self._as_reso(<NPY_DATETIMEUNIT>(self._reso + 1))
+            except OverflowError:
```

Comment:
Do the unit test hit both the `except` and `else` branch?

Reply:
just reach one branch, will add test(s)

Reply:
Did these get added?

Reply:
good catch, not yet

Reply:
updated per request